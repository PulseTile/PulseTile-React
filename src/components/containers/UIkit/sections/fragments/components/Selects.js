import React, { Component } from 'react';

const themes = [
  { id: 'green',  name: 'Green Theme',  baseColor: '#0D672F' },
  { id: 'red',    name: 'Red Theme',    baseColor: 'red'     },
  { id: 'blue',   name: 'Blue Theme',   baseColor: 'blue'    },
  { id: 'brown',  name: 'Brown Theme',  baseColor: 'brown'   },
  { id: 'yellow', name: 'Yellow Theme', baseColor: 'yellow'  },
];

/**
 * This component returns content of Selects section
 *
 * @return {XML}
 * @constructor
 */
export default class Selects extends Component {

  state = {
    currentId: 'green',
    isDropdownOpen: false,
  };

  /**
   * This function opens dropdown of the Theme selector
   */
  openDropdown = () => {
    this.setState({
      isDropdownOpen: !this.state.isDropdownOpen,
    });
  };

  /**
   * This function changes current theme
   */
  changeTheme = id => {
    this.setState({
      currentId: id,
    });
  };

  /**
   * This function returns current theme
   */
  getCurrentTheme = currentId => {
    let result = null;
    for (let i = 0, n = themes.length; i < n; i++) {
      let item = themes[i];
      if (item.id === currentId) {
        result = item;
        break;
      }
    }
    return result;
  };

  render() {
    const { isDropdownOpen, currentId } = this.state;
    const currentTheme = this.getCurrentTheme(currentId);
    return (
      <div id="select" className="ui-section">
        <strong className="ui-title">Selects</strong>
          <div className="form-group-wrapper">
            <div className="row">
              <div className="col-xs-12 col-sm-6">
                <div className="form-group">
                  <label className="control-label">Select</label>
                  <div className="input-holder">
                    <select className="form-control input-sm" id="select1" name="select1">
                      <option selected>Select 1</option>
                      <option>Select 2</option>
                      <option>Select 3</option>
                      <option>Select 4</option>
                      <option>Select 5</option>
                      <option>Select 6</option>
                    </select>
                  </div>
                </div>
              </div>
              <div className="col-xs-12 col-sm-6">
                <div className="form-group">
                  <label className="control-label">Select (disabled)</label>
                  <div className="input-holder">
                    <select className="form-control input-sm" id="select2" name="select2" disabled>
                      <option selected>Select 1</option>
                      <option>Select 2</option>
                      <option>Select 3</option>
                      <option>Select 4</option>
                      <option>Select 5</option>
                      <option>Select 6</option>
                    </select>
                  </div>
                </div>
              </div>
            </div>
            <div className="row">
              <div className="col-xs-12 col-sm-6">
                <div className="form-group">
                  <label className="control-label">Custom Select</label>
                  <div className="input-holder">
                    <div className="input-holder">
                      <div className="ui-select-container ui-select-bootstrap dropdown open" onClick={() => this.openDropdown()}>
                        <div className="ui-select-match">
                          <span className="btn btn-default form-control ui-select-toggle">
                            <span className="ui-select-placeholder text-muted"></span>
                            <span className="ui-select-match-text pull-left">
                              <div className="palette-color">
                                <div className="palette-color-icon" style={{background: currentTheme.baseColor}}></div>
                                <div className="palette-color-name">{currentTheme.name}</div>
                              </div>
                            </span>
                          </span>
                        </div>
                        { isDropdownOpen &&
                          <ul className="ui-select-choices ui-select-choices-content ui-select-dropdown dropdown-menu">
                            { themes.map((item, key) => {
                              return (
                                <li key={key} className="ui-select-choices-group">
                                  <div className="ui-select-match" onClick={() => this.changeTheme(item.id)}>
                                    <span className="btn btn-default form-control ui-select-toggle">
                                      <span className="ui-select-placeholder text-muted"></span>
                                      <span className="ui-select-match-text pull-left">
                                        <div className="palette-color">
                                          <div className="palette-color-icon" style={{background: item.baseColor}}></div>
                                          <div className="palette-color-name">{item.name}</div>
                                        </div>
                                      </span>
                                    </span>
                                  </div>
                                </li>
                              );
                             })}
                          </ul>
                        }
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div className="col-xs-12 col-sm-6">
                <div className="form-group">
                  <label className="control-label">Select (disabled)</label>
                  <div className="input-holder">
                    <div className="ui-select-container ui-select-bootstrap dropdown open" disabled="disabled">
                      <div className="ui-select-match" disabled="disabled">
                        <span className="btn btn-default form-control ui-select-toggle" disabled="disabled">
                          <span className="ui-select-placeholder text-muted"></span>
                          <span className="ui-select-match-text pull-left">
                            <div className="palette-color">
                              <div className="palette-color-icon"></div>
                              <div className="palette-color-name">Green Theme</div>
                            </div>
                          </span>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
    );
  }
}
