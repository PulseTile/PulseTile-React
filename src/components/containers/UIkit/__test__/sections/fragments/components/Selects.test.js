import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import Selects from '../../../../sections/fragments/components/Selects';

Enzyme.configure({ adapter: new Adapter() });

describe('Component <Selects />', () => {
  it('should renders Selects with props correctly', () => {
    const component = shallow(<Selects />);
    expect(component).toMatchSnapshot();

    component.find('.ui-select-container.ui-select-bootstrap.dropdown').at(0).simulate('click');
    expect(component).toMatchSnapshot();
    component.find('.ui-select-container.ui-select-bootstrap.dropdown').at(0).simulate('click');

    component.setState({ isDropdownOpen: true });
    component.find('li .ui-select-match').at(0).simulate('click');
    expect(component).toMatchSnapshot();
  });
});
