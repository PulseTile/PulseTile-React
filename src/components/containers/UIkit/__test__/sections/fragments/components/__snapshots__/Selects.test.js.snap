// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <Selects /> should renders Selects with props correctly 1`] = `
<div
  className="ui-section"
  id="select"
>
  <strong
    className="ui-title"
  >
    Selects
  </strong>
  <div
    className="form-group-wrapper"
  >
    <div
      className="row"
    >
      <div
        className="col-xs-12 col-sm-6"
      >
        <div
          className="form-group"
        >
          <label
            className="control-label"
          >
            Select
          </label>
          <div
            className="input-holder"
          >
            <select
              className="form-control input-sm"
              id="select1"
              name="select1"
            >
              <option
                selected={true}
              >
                Select 1
              </option>
              <option>
                Select 2
              </option>
              <option>
                Select 3
              </option>
              <option>
                Select 4
              </option>
              <option>
                Select 5
              </option>
              <option>
                Select 6
              </option>
            </select>
          </div>
        </div>
      </div>
      <div
        className="col-xs-12 col-sm-6"
      >
        <div
          className="form-group"
        >
          <label
            className="control-label"
          >
            Select (disabled)
          </label>
          <div
            className="input-holder"
          >
            <select
              className="form-control input-sm"
              disabled={true}
              id="select2"
              name="select2"
            >
              <option
                selected={true}
              >
                Select 1
              </option>
              <option>
                Select 2
              </option>
              <option>
                Select 3
              </option>
              <option>
                Select 4
              </option>
              <option>
                Select 5
              </option>
              <option>
                Select 6
              </option>
            </select>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-xs-12 col-sm-6"
      >
        <div
          className="form-group"
        >
          <label
            className="control-label"
          >
            Custom Select
          </label>
          <div
            className="input-holder"
          >
            <div
              className="input-holder"
            >
              <div
                className="ui-select-container ui-select-bootstrap dropdown open"
                onClick={[Function]}
              >
                <div
                  className="ui-select-match"
                >
                  <span
                    className="btn btn-default form-control ui-select-toggle"
                  >
                    <span
                      className="ui-select-placeholder text-muted"
                    />
                    <span
                      className="ui-select-match-text pull-left"
                    >
                      <div
                        className="palette-color"
                      >
                        <div
                          className="palette-color-icon"
                          style={
                            Object {
                              "background": "#0D672F",
                            }
                          }
                        />
                        <div
                          className="palette-color-name"
                        >
                          Green Theme
                        </div>
                      </div>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-xs-12 col-sm-6"
      >
        <div
          className="form-group"
        >
          <label
            className="control-label"
          >
            Select (disabled)
          </label>
          <div
            className="input-holder"
          >
            <div
              className="ui-select-container ui-select-bootstrap dropdown open"
              disabled="disabled"
            >
              <div
                className="ui-select-match"
                disabled="disabled"
              >
                <span
                  className="btn btn-default form-control ui-select-toggle"
                  disabled="disabled"
                >
                  <span
                    className="ui-select-placeholder text-muted"
                  />
                  <span
                    className="ui-select-match-text pull-left"
                  >
                    <div
                      className="palette-color"
                    >
                      <div
                        className="palette-color-icon"
                      />
                      <div
                        className="palette-color-name"
                      >
                        Green Theme
                      </div>
                    </div>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component <Selects /> should renders Selects with props correctly 2`] = `
<div
  className="ui-section"
  id="select"
>
  <strong
    className="ui-title"
  >
    Selects
  </strong>
  <div
    className="form-group-wrapper"
  >
    <div
      className="row"
    >
      <div
        className="col-xs-12 col-sm-6"
      >
        <div
          className="form-group"
        >
          <label
            className="control-label"
          >
            Select
          </label>
          <div
            className="input-holder"
          >
            <select
              className="form-control input-sm"
              id="select1"
              name="select1"
            >
              <option
                selected={true}
              >
                Select 1
              </option>
              <option>
                Select 2
              </option>
              <option>
                Select 3
              </option>
              <option>
                Select 4
              </option>
              <option>
                Select 5
              </option>
              <option>
                Select 6
              </option>
            </select>
          </div>
        </div>
      </div>
      <div
        className="col-xs-12 col-sm-6"
      >
        <div
          className="form-group"
        >
          <label
            className="control-label"
          >
            Select (disabled)
          </label>
          <div
            className="input-holder"
          >
            <select
              className="form-control input-sm"
              disabled={true}
              id="select2"
              name="select2"
            >
              <option
                selected={true}
              >
                Select 1
              </option>
              <option>
                Select 2
              </option>
              <option>
                Select 3
              </option>
              <option>
                Select 4
              </option>
              <option>
                Select 5
              </option>
              <option>
                Select 6
              </option>
            </select>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-xs-12 col-sm-6"
      >
        <div
          className="form-group"
        >
          <label
            className="control-label"
          >
            Custom Select
          </label>
          <div
            className="input-holder"
          >
            <div
              className="input-holder"
            >
              <div
                className="ui-select-container ui-select-bootstrap dropdown open"
                onClick={[Function]}
              >
                <div
                  className="ui-select-match"
                >
                  <span
                    className="btn btn-default form-control ui-select-toggle"
                  >
                    <span
                      className="ui-select-placeholder text-muted"
                    />
                    <span
                      className="ui-select-match-text pull-left"
                    >
                      <div
                        className="palette-color"
                      >
                        <div
                          className="palette-color-icon"
                          style={
                            Object {
                              "background": "#0D672F",
                            }
                          }
                        />
                        <div
                          className="palette-color-name"
                        >
                          Green Theme
                        </div>
                      </div>
                    </span>
                  </span>
                </div>
                <ul
                  className="ui-select-choices ui-select-choices-content ui-select-dropdown dropdown-menu"
                >
                  <li
                    className="ui-select-choices-group"
                    key="0"
                  >
                    <div
                      className="ui-select-match"
                      onClick={[Function]}
                    >
                      <span
                        className="btn btn-default form-control ui-select-toggle"
                      >
                        <span
                          className="ui-select-placeholder text-muted"
                        />
                        <span
                          className="ui-select-match-text pull-left"
                        >
                          <div
                            className="palette-color"
                          >
                            <div
                              className="palette-color-icon"
                              style={
                                Object {
                                  "background": "#0D672F",
                                }
                              }
                            />
                            <div
                              className="palette-color-name"
                            >
                              Green Theme
                            </div>
                          </div>
                        </span>
                      </span>
                    </div>
                  </li>
                  <li
                    className="ui-select-choices-group"
                    key="1"
                  >
                    <div
                      className="ui-select-match"
                      onClick={[Function]}
                    >
                      <span
                        className="btn btn-default form-control ui-select-toggle"
                      >
                        <span
                          className="ui-select-placeholder text-muted"
                        />
                        <span
                          className="ui-select-match-text pull-left"
                        >
                          <div
                            className="palette-color"
                          >
                            <div
                              className="palette-color-icon"
                              style={
                                Object {
                                  "background": "red",
                                }
                              }
                            />
                            <div
                              className="palette-color-name"
                            >
                              Red Theme
                            </div>
                          </div>
                        </span>
                      </span>
                    </div>
                  </li>
                  <li
                    className="ui-select-choices-group"
                    key="2"
                  >
                    <div
                      className="ui-select-match"
                      onClick={[Function]}
                    >
                      <span
                        className="btn btn-default form-control ui-select-toggle"
                      >
                        <span
                          className="ui-select-placeholder text-muted"
                        />
                        <span
                          className="ui-select-match-text pull-left"
                        >
                          <div
                            className="palette-color"
                          >
                            <div
                              className="palette-color-icon"
                              style={
                                Object {
                                  "background": "blue",
                                }
                              }
                            />
                            <div
                              className="palette-color-name"
                            >
                              Blue Theme
                            </div>
                          </div>
                        </span>
                      </span>
                    </div>
                  </li>
                  <li
                    className="ui-select-choices-group"
                    key="3"
                  >
                    <div
                      className="ui-select-match"
                      onClick={[Function]}
                    >
                      <span
                        className="btn btn-default form-control ui-select-toggle"
                      >
                        <span
                          className="ui-select-placeholder text-muted"
                        />
                        <span
                          className="ui-select-match-text pull-left"
                        >
                          <div
                            className="palette-color"
                          >
                            <div
                              className="palette-color-icon"
                              style={
                                Object {
                                  "background": "brown",
                                }
                              }
                            />
                            <div
                              className="palette-color-name"
                            >
                              Brown Theme
                            </div>
                          </div>
                        </span>
                      </span>
                    </div>
                  </li>
                  <li
                    className="ui-select-choices-group"
                    key="4"
                  >
                    <div
                      className="ui-select-match"
                      onClick={[Function]}
                    >
                      <span
                        className="btn btn-default form-control ui-select-toggle"
                      >
                        <span
                          className="ui-select-placeholder text-muted"
                        />
                        <span
                          className="ui-select-match-text pull-left"
                        >
                          <div
                            className="palette-color"
                          >
                            <div
                              className="palette-color-icon"
                              style={
                                Object {
                                  "background": "yellow",
                                }
                              }
                            />
                            <div
                              className="palette-color-name"
                            >
                              Yellow Theme
                            </div>
                          </div>
                        </span>
                      </span>
                    </div>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-xs-12 col-sm-6"
      >
        <div
          className="form-group"
        >
          <label
            className="control-label"
          >
            Select (disabled)
          </label>
          <div
            className="input-holder"
          >
            <div
              className="ui-select-container ui-select-bootstrap dropdown open"
              disabled="disabled"
            >
              <div
                className="ui-select-match"
                disabled="disabled"
              >
                <span
                  className="btn btn-default form-control ui-select-toggle"
                  disabled="disabled"
                >
                  <span
                    className="ui-select-placeholder text-muted"
                  />
                  <span
                    className="ui-select-match-text pull-left"
                  >
                    <div
                      className="palette-color"
                    >
                      <div
                        className="palette-color-icon"
                      />
                      <div
                        className="palette-color-name"
                      >
                        Green Theme
                      </div>
                    </div>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component <Selects /> should renders Selects with props correctly 3`] = `
<div
  className="ui-section"
  id="select"
>
  <strong
    className="ui-title"
  >
    Selects
  </strong>
  <div
    className="form-group-wrapper"
  >
    <div
      className="row"
    >
      <div
        className="col-xs-12 col-sm-6"
      >
        <div
          className="form-group"
        >
          <label
            className="control-label"
          >
            Select
          </label>
          <div
            className="input-holder"
          >
            <select
              className="form-control input-sm"
              id="select1"
              name="select1"
            >
              <option
                selected={true}
              >
                Select 1
              </option>
              <option>
                Select 2
              </option>
              <option>
                Select 3
              </option>
              <option>
                Select 4
              </option>
              <option>
                Select 5
              </option>
              <option>
                Select 6
              </option>
            </select>
          </div>
        </div>
      </div>
      <div
        className="col-xs-12 col-sm-6"
      >
        <div
          className="form-group"
        >
          <label
            className="control-label"
          >
            Select (disabled)
          </label>
          <div
            className="input-holder"
          >
            <select
              className="form-control input-sm"
              disabled={true}
              id="select2"
              name="select2"
            >
              <option
                selected={true}
              >
                Select 1
              </option>
              <option>
                Select 2
              </option>
              <option>
                Select 3
              </option>
              <option>
                Select 4
              </option>
              <option>
                Select 5
              </option>
              <option>
                Select 6
              </option>
            </select>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-xs-12 col-sm-6"
      >
        <div
          className="form-group"
        >
          <label
            className="control-label"
          >
            Custom Select
          </label>
          <div
            className="input-holder"
          >
            <div
              className="input-holder"
            >
              <div
                className="ui-select-container ui-select-bootstrap dropdown open"
                onClick={[Function]}
              >
                <div
                  className="ui-select-match"
                >
                  <span
                    className="btn btn-default form-control ui-select-toggle"
                  >
                    <span
                      className="ui-select-placeholder text-muted"
                    />
                    <span
                      className="ui-select-match-text pull-left"
                    >
                      <div
                        className="palette-color"
                      >
                        <div
                          className="palette-color-icon"
                          style={
                            Object {
                              "background": "#0D672F",
                            }
                          }
                        />
                        <div
                          className="palette-color-name"
                        >
                          Green Theme
                        </div>
                      </div>
                    </span>
                  </span>
                </div>
                <ul
                  className="ui-select-choices ui-select-choices-content ui-select-dropdown dropdown-menu"
                >
                  <li
                    className="ui-select-choices-group"
                    key="0"
                  >
                    <div
                      className="ui-select-match"
                      onClick={[Function]}
                    >
                      <span
                        className="btn btn-default form-control ui-select-toggle"
                      >
                        <span
                          className="ui-select-placeholder text-muted"
                        />
                        <span
                          className="ui-select-match-text pull-left"
                        >
                          <div
                            className="palette-color"
                          >
                            <div
                              className="palette-color-icon"
                              style={
                                Object {
                                  "background": "#0D672F",
                                }
                              }
                            />
                            <div
                              className="palette-color-name"
                            >
                              Green Theme
                            </div>
                          </div>
                        </span>
                      </span>
                    </div>
                  </li>
                  <li
                    className="ui-select-choices-group"
                    key="1"
                  >
                    <div
                      className="ui-select-match"
                      onClick={[Function]}
                    >
                      <span
                        className="btn btn-default form-control ui-select-toggle"
                      >
                        <span
                          className="ui-select-placeholder text-muted"
                        />
                        <span
                          className="ui-select-match-text pull-left"
                        >
                          <div
                            className="palette-color"
                          >
                            <div
                              className="palette-color-icon"
                              style={
                                Object {
                                  "background": "red",
                                }
                              }
                            />
                            <div
                              className="palette-color-name"
                            >
                              Red Theme
                            </div>
                          </div>
                        </span>
                      </span>
                    </div>
                  </li>
                  <li
                    className="ui-select-choices-group"
                    key="2"
                  >
                    <div
                      className="ui-select-match"
                      onClick={[Function]}
                    >
                      <span
                        className="btn btn-default form-control ui-select-toggle"
                      >
                        <span
                          className="ui-select-placeholder text-muted"
                        />
                        <span
                          className="ui-select-match-text pull-left"
                        >
                          <div
                            className="palette-color"
                          >
                            <div
                              className="palette-color-icon"
                              style={
                                Object {
                                  "background": "blue",
                                }
                              }
                            />
                            <div
                              className="palette-color-name"
                            >
                              Blue Theme
                            </div>
                          </div>
                        </span>
                      </span>
                    </div>
                  </li>
                  <li
                    className="ui-select-choices-group"
                    key="3"
                  >
                    <div
                      className="ui-select-match"
                      onClick={[Function]}
                    >
                      <span
                        className="btn btn-default form-control ui-select-toggle"
                      >
                        <span
                          className="ui-select-placeholder text-muted"
                        />
                        <span
                          className="ui-select-match-text pull-left"
                        >
                          <div
                            className="palette-color"
                          >
                            <div
                              className="palette-color-icon"
                              style={
                                Object {
                                  "background": "brown",
                                }
                              }
                            />
                            <div
                              className="palette-color-name"
                            >
                              Brown Theme
                            </div>
                          </div>
                        </span>
                      </span>
                    </div>
                  </li>
                  <li
                    className="ui-select-choices-group"
                    key="4"
                  >
                    <div
                      className="ui-select-match"
                      onClick={[Function]}
                    >
                      <span
                        className="btn btn-default form-control ui-select-toggle"
                      >
                        <span
                          className="ui-select-placeholder text-muted"
                        />
                        <span
                          className="ui-select-match-text pull-left"
                        >
                          <div
                            className="palette-color"
                          >
                            <div
                              className="palette-color-icon"
                              style={
                                Object {
                                  "background": "yellow",
                                }
                              }
                            />
                            <div
                              className="palette-color-name"
                            >
                              Yellow Theme
                            </div>
                          </div>
                        </span>
                      </span>
                    </div>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-xs-12 col-sm-6"
      >
        <div
          className="form-group"
        >
          <label
            className="control-label"
          >
            Select (disabled)
          </label>
          <div
            className="input-holder"
          >
            <div
              className="ui-select-container ui-select-bootstrap dropdown open"
              disabled="disabled"
            >
              <div
                className="ui-select-match"
                disabled="disabled"
              >
                <span
                  className="btn btn-default form-control ui-select-toggle"
                  disabled="disabled"
                >
                  <span
                    className="ui-select-placeholder text-muted"
                  />
                  <span
                    className="ui-select-match-text pull-left"
                  >
                    <div
                      className="palette-color"
                    >
                      <div
                        className="palette-color-icon"
                      />
                      <div
                        className="palette-color-name"
                      >
                        Green Theme
                      </div>
                    </div>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
